graylog2-web-interface/src/preflight/theme/ThemeWrapper.tsx

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import React, { useContext } from 'react';
+import { ThemeContext } from 'styled-components';
+import type { MantineThemeOverride } from '@mantine/core';
+import { Global, MantineProvider } from '@mantine/core';
+
+type Props = {
+  children: React.ReactElement,
+};
+
+const ThemeWrapper = ({ children }: Props) => {
+  const theme = useContext(ThemeContext);
+
+  const mantineTheme: MantineThemeOverride = {
+    colorScheme: theme.mode === 'teint' ? 'light' : 'dark',
+    fontFamily: theme.fonts.family.body,
+    fontFamilyMonospace: theme.fonts.family.monospace,
+    fontSizes: {
+      xs: theme.fonts.size.tiny.value,
+      sm: theme.fonts.size.small.value,
+      md: theme.fonts.size.body.value,
+      lg: theme.fonts.size.large.value,
+      xl: theme.fonts.size.huge.value,
+    },
+    headings: {
+      fontFamily: theme.fonts.family.body,
+      sizes: {
+        h1: {
+          fontSize: theme.fonts.size.h1.rem,
+        },
+        h2: {
+          fontSize: theme.fonts.size.h2.rem,
+        },
+        h3: {
+          fontSize: theme.fonts.size.h3.rem,
+        },
+        h4: {
+          fontSize: theme.fonts.size.h4.rem,
+        },
+        h5: {
+          fontSize: theme.fonts.size.h5.rem,
+        },
+        h6: {
+          fontSize: theme.fonts.size.h6.rem,
+        },
+      },
+    },
+    spacing: {
+      xs: theme.spacings.px.xs,
+      sm: theme.spacings.px.sm,
+      lg: theme.spacings.px.lg,
+      xl: theme.spacings.px.xl,
+    },
```

Comment:
It looks like this version of mantine now expects the spacings to have a unit.

Instead we can use:
```
xs: theme.spacings.xs,
sm: theme.spacings.sm,
lg: theme.spacings.lg,
xl: theme.spacings.xl,
```

Reply:
I am not sure why we do not see a TS error here.
Having a look at the type definition, TS expects the object values to be a string:

<img width="399" alt="image" src="https://user-images.githubusercontent.com/46300478/231424769-88916eb7-8afe-4ae8-bbab-32fa2a60246a.png">

But we actually define a number:
<img width="350" alt="image" src="https://user-images.githubusercontent.com/46300478/231424977-7f321c69-f216-42fd-833b-31c22d405164.png">